src/coreclr/jit/emitarm64.h

```diff
@@ -576,6 +578,40 @@ static code_t insEncodeSveElemsize_dtype(instruction ins, emitAttr size, code_t
 // for the 'dtype' field.
 static code_t insEncodeSveElemsize_dtype_ld1w(instruction ins, insFormat fmt, emitAttr size, code_t code);
 
+// Encodes an immediate value in consecutive bits from most signficant position 'hi' to least significant
+// position 'lo'.
+template <const size_t hi, const size_t lo>
+static code_t insEncodeUimm(size_t imm)
```

Comment:
We tend to write explicit `insEncodeUimm*`, but I'm not opposed to have something more generalized.
@BruceForstall @kunalspathak what are the guidelines for writing generalized functions like this?

Reply:
I like this idea. Thanks @snickolls-arm . Will it be possible to replace existing methods with this one and make sure you don't hit any asserts?

Reply:
Yes I'll take a look at replacing these. Would you prefer this in a separate patch or on this one?

Reply:
separate path is fine.